gson/src/test/java/com/google/gson/JsonParserTest.java

```diff
@@ -19,12 +19,14 @@
 import java.io.CharArrayReader;
```

Comment:
Might be good to add a test for the non-execute prefix as well.

Reply:
It seems that there are already [unit tests](https://github.com/google/gson/blob/ceae88bd6667f4263bbe02e6b3710b8a683906a2/gson/src/test/java/com/google/gson/stream/JsonReaderTest.java#L1283) for the non-executable prefix management. Could you elaborate a little?

Reply:
Yes, sorry for not being very clear. I meant a new test for a non-execute prefix followed by EOF (which your code now supports).

Reply:
> to make sure that a non-execute prefix followed by EOF still throws an EOFException

Wouldn't we _not_ want an exception in such case? (I assume you meant from using `JsonReader::peek()`).
In strict mode, non-execute prefix won't be accepted.
In lenient mode, it will be ignored and the together following EOF will be interpreted as an empty document.

Reply:
It is probably subjective what the _right_ behavior is here. I thought if a document contains a non execute prefix it is non-empty (even though it contains no JSON tokens). Though on the other hand a third party library which always writes the non execute prefix and then checks whether there is actually JSON data to write, might consider a non execute prefix followed by EOF to be an empty document.

Reply:
Yeah I agree it does seem subjective.

Based on my understanding, the current behavior is what I described in my comment above. Essentially the current implementation in `Stream::parse` uses `EOFException` to determine EOF, whereas my PR makes it an explicit check on what `JsonReader::peek` returns, without throwing `EOFException`.

I thought `EOFException` should be used when detecting EOF in the middle of parsing a JSON value, but not when the document (excluding non-execute prefix in lenient mode) is empty.